### DISCUSSION TITLE
Respect Client API addCustomFilter & addPreSearch

### DISCUSSION THREAD

**Discussion by EdgarasKarpavicius** [drivardxrm/LookupDropdown.PCF#19] [Ideas]
Hi David,

Firstly, thanks for making this available to the community. I have used it across various projects and is a great, yet subtle addition to any model-driven app out there for when the use-case is right.

I have hit a requirement where a I need to filter a lookup table based on an option set field already selected on the same form, and naturally looked to use the client api .addCustomFilter method for the onChange and onLoad events for the form [More info here.](https://learn.microsoft.com/en-us/power-apps/developer/model-driven-apps/clientapi/reference/controls/addcustomfilter) 

While this works great with the default lookup control, and applies a filter to the list of available records to select from based on an optionset value already selected, unfortunately when I swap the field to use your dropdown control (instead of the default), it does not appear to respect the filtering in the same manner, returning the full list of records from the specified view, ignoring the additional filters.

I just wanted to make you aware, and ask if there would be any opportunity for your component to meet this functionality. I appreciate this can be time consuming, so we could give a hand from our side too.

Kind Regards
Edgar

**Comment by EdgarasKarpavicius** [comment]
Edit: just seen the same item raised under 'Issues' back in April. Please feel free to delete this discussion.

**Comment by drivardxrm** [comment]
Hi @EdgarasKarpavicius ,
Thanks for your interest in the control.

The ability to honour the customFilter is on the roadmap, but its a bit more complicated than I thought.

I will keep you updated.

Thanks for your feedback.

**Comment by TayloredMatt** [comment]
Hi @drivardxrm, 

Loving the control - something I've been hoping to find as it was a bit beyond me creating it myself. I am just wondering whether you had any luck with adding the custom filters, as I've tried today and couldn't get anything to work.

**Reply by drivardxrm** [reply]
Unfortunately, I haven't had the time to implement the feature yet
